### DISCUSSION TITLE
How do I set the DNS propagation timeout?

### DISCUSSION THREAD

**Discussion by appliedprivacy** [go-acme/lego#1882] [Q&A] [answered]
Hi,

when renewing certificates using DNS challenge with DeSEC.io 
lego waits for up to 1min to see the DNS record:

```
[INFO] Wait for propagation [timeout: 1m0s, interval: 2s]
```

Unfortunately DNS propagation takes slightly longer - maybe 70 seconds.

I did not find any CLI switches to increase the 1min timeout. Is there one? or is there an env var for it? 

Or is this timeout hardcoded per DNS provider?
https://go-acme.github.io/lego/usage/cli/options/

`--dns-timeout` is apparently not the timeout I'm trying to change.

**Comment by ldez** [comment, marked as answer]
Hi,

There are no CLI flags for DNS providers, the configuration is handled per providers and through env vars.

https://go-acme.github.io/lego/dns/desec/
